### PR TITLE
Fix addressbook erratic navigation behavior

### LINKED ISSUE #1063: Navigating the address book menus sometimes fails to update the view
https://github.com/DarkFlorist/TheInterceptor/assets/1169838/cfc6c226-4b6e-4c3e-8d07-d863e9385c75

### PR DESCRIPTION
**Pull request by jubalm**
Fixes #1063

### CODE REVIEW

File: app/ts/components/subcomponents/DynamicScroller.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		const containerObserver = new ResizeObserver(([entry]) => {
+			maxItems.value = Math.ceil(entry!.contentRect.height / itemHeight.value)
+		})
+		containerObserver.observe(scrollViewRef.current)
+		return () => { containerObserver.disconnect() }
+	})
+
+	return (
+		<div ref = { scrollViewRef } style = { { overflowY: 'scroll', maxHeight: '100%' } } onScroll = { recalculateStartIndex }>
+			<div style = { { height: `${scrollAreaHeight}px`, '--virtual-scroll-offset': `${scrollOffset}px` } }>
```
Comment (KillariDev) [inline app/ts/components/subcomponents/DynamicScroller.tsx:46]:
```suggestion
			<div style = { { height: `${ scrollAreaHeight }px`, '--virtual-scroll-offset': `${ scrollOffset }px` } }>
```

**Review by KillariDev — COMMENTED** [review]
Good changes! I found couple bugs:

there's double scrollbar now:
image

the x:s are sometimes with blue border (when you cannot remove them, and I feel blue is not the right way to indicate that the button is disabled):
image

**Review by KillariDev — APPROVED** [review]

**Comment by KillariDev** [comment]
I guess this fixes this? https://github.com/DarkFlorist/TheInterceptor/issues/1063

**Comment by jubalm** [comment]
> I guess this fixes this? #1063

**Comment by jubalm** [comment]
> Good changes! I found couple bugs:
> 
> there's double scrollbar now: image
> 
> the x:s are sometimes with blue border (when you cannot remove them, and I feel blue is not the right way to indicate that the button is disabled): image

double scrollbar should be fixed with [c9a443b](https://github.com/DarkFlorist/TheInterceptor/pull/1064/commits/c9a443b3699ceb01530c431b21ca440aa7edae5f)

**Pull request merged**